src/utils/address.ts

```diff
@@ -76,4 +78,17 @@ export async function deriveAddressP2SH(index: number, storage: IStorage): Promi
     throw new Error('No multisig data');
   }
   return deriveAddressFromDataP2SH(multisigData, index, storage.config.getNetwork().name);
-}
\ No newline at end of file
+}
+
+/**
+ * Parse the public key and return an address.
+ *
+ * @param pubkey Hex string conveying the public key.
+ * @param network Address's network.
+ * @returns The address object from parsed publicKey
+ */
+export function getAddressFromPubkey(pubkey: string, network: Network): Address {
```

Comment:
This method will work only for P2PKH, if I'm not wrong. If the goal of this method is to be used to parse nc caller, we should add support for the P2SH addresses as well.

Reply:
Oh. We need to talk them.

Reply:
As we don't support P2SH on wallet-service and other scenarios, we can discuss it in the future.